storm-core/src/jvm/org/apache/storm/daemon/Acker.java

```diff
@@ -100,6 +101,8 @@ public void execute(Tuple input) {
             }
             curr.failed = true;
             pending.put(id, curr);
+        } else if(ACKER_RESET_TIMEOUT_STREAM_ID.equals(streamId)) {
```

Comment:
This introduces a NPE scenario because earlier in all the if blocks, if curr was null, it was getting initialized. Now curr.spoutTask will throw NPE if reset timeout comes before init or after init times out. This can be avoided by adding below just before `pending.put(id, curr)`
```
if (curr == null) {
     curr = new AckObject();
}
```